src/modules/navigator/MissionFeasibility/FeasibilityChecker.cpp

```diff
@@ -0,0 +1,740 @@
+					     "First waypoint too far away: %dm, %d max\t",
+					     (int)dist_to_1wp, (int)_param_mis_dist_1wp);
+			events::send<uint32_t, uint32_t>(events::ID("navigator_mis_first_wp_too_far"), {events::Log::Error, events::LogInternal::Info},
+							 "First waypoint too far away: {1m} (maximum: {2m})", (uint32_t)dist_to_1wp, (uint32_t)_param_mis_dist_1wp);
+
+			return false;
+		}
+	}
+
+	/* no waypoints found in mission, then we will not fly far away */
+	return true;
+}
+
+bool FeasibilityChecker::checkDistancesBetweenWaypoints(const mission_item_s &mission_item)
+{
+	if (_param_mis_dist_wps <= 0.0f) {
+		/* param not set, check is ok */
+		return true;
+	}
+
+
+	/* check only items with valid lat/lon */
+	if (!MissionBlock::item_contains_position(mission_item)) {
+		return true;
+	}
+
+	/* Compare it to last waypoint if already available. */
+	if (PX4_ISFINITE(_last_lat) && PX4_ISFINITE(_last_lon)) {
+		/* check distance from current position to item */
+		const float dist_between_waypoints = get_distance_to_next_waypoint(
+				mission_item.lat, mission_item.lon,
+				_last_lat, _last_lon);
+
+
+		if (dist_between_waypoints > _param_mis_dist_wps) {
+			/* distance between waypoints is too high */
+			mavlink_log_critical(_mavlink_log_pub,
+					     "Distance between waypoints too far: %d meters, %d max.\t",
+					     (int)dist_between_waypoints, (int)_param_mis_dist_wps);
+			events::send<uint32_t, uint32_t>(events::ID("navigator_mis_wp_dist_too_far"), {events::Log::Error, events::LogInternal::Info},
+							 "Distance between waypoints too far: {1m}, (maximum: {2m})", (uint32_t)dist_between_waypoints,
+							 (uint32_t)_param_mis_dist_wps);
+
+
+			return false;
+
+			/* do not allow waypoints that are literally on top of each other */
+
+			/* and do not allow condition gates that are at the same position as a navigation waypoint */
+
+		} else if (dist_between_waypoints < 0.05f &&
+			   (mission_item.nav_cmd == NAV_CMD_CONDITION_GATE || _last_cmd == NAV_CMD_CONDITION_GATE)) {
+
+			/* Waypoints and gate are at the exact same position, which indicates an
+			 * invalid mission and makes calculating the direction from one waypoint
+			 * to another impossible. */
+			mavlink_log_critical(_mavlink_log_pub,
+					     "Distance between waypoint and gate too close: %d meters\t",
+					     (int)dist_between_waypoints);
+			events::send<float, float>(events::ID("navigator_mis_wp_gate_too_close"), {events::Log::Error, events::LogInternal::Info},
+						   "Distance between waypoint and gate too close: {1:.3m} (minimum: {2:.3m})", dist_between_waypoints, 0.05f);
+
+
+			return false;
+		}
+	}
+
+	_last_lat = mission_item.lat;
+	_last_lon = mission_item.lon;
+	_last_cmd = mission_item.nav_cmd;
+
+	/* We ran through all waypoints and have not found any distances between waypoints that are too far. */
+	return true;
+}
+
+bool FeasibilityChecker::checkIfBelowHomeAltitude(const mission_item_s &mission_item, const int current_index)
```

Comment:
That may also be one to get rid of, Missions below Home are in the end totally feasible. Or maybe it could be a simple warning, but not prevent it completely?

Reply:
okay great